src/value.c

```diff
@@ -498,10 +498,31 @@ static SIValue SIValue_ConcatList(const SIValue a, const SIValue b) {
 	return resultArray;
 }
 
+// assumption: a and b are maps
+static SIValue SIValue_MergeMap(const SIValue a, const SIValue b) {
+	SIValue result = Map_Clone(a);
+
+	// Merge b into result
+	uint bLen = Map_KeyCount(b);
+	for(uint i = 0; i < bLen; i++) {
+		SIValue key, value;
+		Map_GetIdx(b, i, &key, &value);
+		Map_Add(&result, key, value);
+	}
+
+	return result;
+}
+
 SIValue SIValue_Add(const SIValue a, const SIValue b) {
 	if(a.type == T_NULL || b.type == T_NULL) return SI_NullVal();
 	if(a.type == T_ARRAY || b.type == T_ARRAY) return SIValue_ConcatList(a, b);
 	if(a.type == T_STRING || b.type == T_STRING) return SIValue_ConcatString(a, b);
+	if(a.type == T_MAP && b.type == T_MAP) return Map_Merge(a, b);
+	if(a.type == T_MAP || b.type == T_MAP) {
+		// one of the operands is a map and the other is not
+		// return NULL
+		return SI_NullVal();
+	}
```

Comment:
Do we have a test which covers this case?
Is that the expected output, e.g. when performing `{a:1} + 2` do we want to get `NULL` or should we emit an error ?

Reply:
changed it to raise error
added the test